File: C++/프로그래머스/1/42840. 모의고사/모의고사.cpp
#include <vector>
#include <algorithm>

using namespace std;

vector<int> solution(vector<int> answers) {
    vector<vector<int>> patterns = {
        {1, 2, 3, 4, 5},
        {2, 1, 2, 3, 2, 4, 2, 5},
        {3, 3, 1, 1, 2, 2, 4, 4, 5, 5}
    };
    
    vector<int> scores(3, 0);
    
    for (size_t i = 0; i < answers.size(); ++i) {
        if (patterns[0][i % patterns[0].size()] == answers[i]) scores[0]++;
        if (patterns[1][i % patterns[1].size()] == answers[i]) scores[1]++;
        if (patterns[2][i % patterns[2].size()] == answers[i]) scores[2]++;
    }
    
    int max_score = *max_element(scores.begin(), scores.end());
    vector<int> result;
    for (int i = 0; i < 3; ++i) {
        if (scores[i] == max_score) result.push_back(i + 1);
    }
    
    return result;
}

File: Python3/프로그래머스/1/42840. 모의고사/모의고사.py
def solution(answers):
    # 각 수포자의 찍는 패턴
    patterns = [
        [1, 2, 3, 4, 5],
        [2, 1, 2, 3, 2, 4, 2, 5],
        [3, 3, 1, 1, 2, 2, 4, 4, 5, 5]
    ]
    
    # 수포자들의 점수를 저장할 리스트
    scores = [0, 0, 0]
    
    # 정답을 비교하여 점수 계산
    for i, answer in enumerate(answers):
        for j, pattern in enumerate(patterns):
            if answer == pattern[i % len(pattern)]:
                scores[j] += 1
    
    # 최대 점수 계산 및 결과 리스트 작성
    max_score = max(scores)
    result = [i + 1 for i, score in enumerate(scores) if score == max_score]
    
    return result
